### PR TITLE
[KYUUBI #2344] [Improvement] Add Kyuubi Server on Kubernetes with Spark Cluster mode integration test

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

Add Kyuubi Server on Kubernetes with Spark Cluster mode integration test

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zwangsheng** [comment]
```
- execute credentials renewal task and wait for completion *** FAILED ***
  The code passed to eventually never returned normally. Attempted 21 times over 1.1136331720000001 seconds. Last failure message: 0 did not equal 1. (HadoopCredentialsManagerSuite.scala:103)
```
CI fail seems nothing to do with this PR, or help retest plz.

File: integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala [hunk 1]
```diff
@@ -17,17 +17,22 @@
...
+
 import io.fabric8.kubernetes.client.DefaultKubernetesClient
 
 import org.apache.kyuubi.WithKyuubiServer
 import org.apache.kyuubi.config.KyuubiConf
 
 trait WithKyuubiServerOnKubernetes extends WithKyuubiServer {
   protected val kyuubiServerConf: KyuubiConf = KyuubiConf()
-  protected val connectionConf: Map[String, String]
+  protected var connectionConf: mutable.Map[String, String] = mutable.Map.empty[String, String]
```
Comment (ulysses-you) [inline integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala:29]: why do we need this change ..

Comment (zwangsheng) [reply integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala:29]: Well, I'm not very familiar with Scala maps, and I'm not sure if += is possible with maps

Comment (zwangsheng) [reply integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala:29]: We need to build duplicate configurations in `KyuubiOnKubernetesWithSparkOnKubernetesTestsBase`, and then build special configurations in `Client test` and `Cluster test`, so we chose not to use a pure Map.

File: integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala [hunk 2]
```diff
@@ -17,16 +17,20 @@
...
 import org.apache.kyuubi.WithKyuubiServer
 import org.apache.kyuubi.config.KyuubiConf
 
 trait WithKyuubiServerOnKubernetes extends WithKyuubiServer {
   protected val kyuubiServerConf: KyuubiConf = KyuubiConf()
-  protected val connectionConf: Map[String, String]
+  protected lazy val connectionConf: mutable.Map[String, String] = setConnectionSparkConf()
   private val miniKubernetesClient: DefaultKubernetesClient = MiniKube.getKubernetesClient
 
+  protected def setConnectionSparkConf(): mutable.Map[String, String] = mutable.Map.empty
```
Comment (ulysses-you) [inline integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala:32]: unnecessary change .. just super. connectionConf ++ childConf

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master
